### PR TITLE
Parse text_source fallbacks

### LINKED ISSUE #607: `text_source` param should account for language fallbacks
Language preference should be able to be defined as a serie of language fallbacks from the text style param `text_source`.

JS reference:
- https://github.com/tangrams/tangram/pull/235

Docs:
- https://github.com/tangrams/tangram-docs/blob/621fecb203c19b91514d544c58330527013cae61/pages/draw.md#text_source

### PR DESCRIPTION
**Pull request by karimnaaji**
Fixes #607

### CODE REVIEW

**Comment by tallytalwar** [comment]
LGTM.

**Comment by tallytalwar** [comment]
@karimnaaji you want to clean the scene.yaml (some commented code, etc), once done will merge.

**Comment by blair1618** [comment]
Has anyone profiled this change with a label-heavy style like eraser-map or cinnabar? I'm a little concerned about the speed of stringstream operations and vector allocation.

**Comment by tallytalwar** [comment]
I do not think there will be anything noticeable big, but profiling is a good call, to confirm.

**Comment by karimnaaji** [comment]
@tallytalwar , I'm good with letting more test cases in the stylesheet/commented blocks that makes it easy to test.
Anyhow we should take some time writing a better default stylesheet, I will create an issue for that.

**Comment by karimnaaji** [comment]
@blair1618 , good call, I will perform some profiling on that.

**Comment by tallytalwar** [comment]
A quick profiling with eraser-map style sheet on master and this branch I get very similar number for `TextStyleBuilder::applyRule` on both branches, about **0.1% (15-20ms)**

**Comment by karimnaaji** [comment]
@tallytalwar , eraser-map stylesheet doesn't use that much this option https://github.com/tangrams/eraser-map/search?utf8=%E2%9C%93&q=text_source. It should be profiled with eraser-map but adding `text_source: [name:en, name, area...]` to more labels.

File: core/src/style/textStyle.cpp [hunk 1]
```diff
@@ -98,14 +102,36 @@ struct TextStyleBuilder : public StyleBuilder {
...
 
+std::string TextStyleBuilder::resolveTextSource(const std::string& textSource, const Properties& props) const {
+
+    std::vector<std::string> fallbacks;
+    std::stringstream ss(textSource);
+    std::string item;
+
+    // Parse fallbacks
+    while (std::getline(ss, item, ',')) {
+        fallbacks.push_back(item);
```
Comment (blair1618) [inline core/src/style/textStyle.cpp]: Can we just check and return the property here? Do we need that `fallbacks` vector?

Comment (karimnaaji) [inline core/src/style/textStyle.cpp]: Updated

**Comment by karimnaaji** [comment]
Profiling didn't show any difference even if using `text_source` option with fallbacks for all eraser-map labels.

**Comment by blair1618** [comment]
A+

Merge when ready!

**Pull request merged**